tools/time_tools.go

```diff
@@ -0,0 +1,24 @@
+package tools
+
+import "time"
+
+// IsExpiredAtOrIn returns whether or not the result of calling TimeAtOrIn is
+// "expired" within "until" units of time from now.
+func IsExpiredAtOrIn(until time.Duration, at time.Time, in time.Duration) (time.Time, bool) {
+	expiration := TimeAtOrIn(at, in)
+	if expiration.IsZero() {
+		return expiration, false
+	}
+
+	return expiration, expiration.Before(time.Now().Add(until))
+}
```

Comment:
I don't think this is correct, since it's always comparing with `time.Now()`. I think the structs would have to store `time.Now()` at the time they're deserialized from the json response. Otherwise a positive `expires_in` value would never expire, right?